### DISCUSSION TITLE
How can I export struct with a function in slint?

### DISCUSSION THREAD

**Discussion by fluolite** [slint-ui/slint#6325] [General]
How can I export struct with a function member in slint, just like export global for rust to callback?
```
export struct Foo {
      bar: string,
      cb:  callback, // can I ?
}

export global UIContext {
    pure callback load_image(string, string,) -> image;
}

```

**Comment by ogoffart** [comment]
struct can't have callback.
Having callback like that wouldn't really work well in slint because if it can have a closure, it would need garbage collection of some sort.

I recommand doing something like this:

```
export struct Foo {
      bar: string,
      id: int,
}

export global UIContext {
   callback foo-callback(int);
}
```

And instead of calling `foo.cb()`  you call `UIContext.foo-callback(foo.id)`

**Reply by fluolite** [reply]
When I has many pages, So I can only be binding it's property and callback in AppWindow ?
